start/src/main/java/org/apache/accumulo/start/classloader/vfs/AccumuloReloadingVFSClassLoader.java

```diff
@@ -199,6 +211,12 @@ public AccumuloReloadingVFSClassLoader(String uris, FileSystemManager vfs,
    * thread
    */
   public void close() {
+    try {
+      Stream.of(this.files).forEach(monitor::removeFile);
+    } catch (RuntimeException re) {
```

Comment:
It might be best to collect all these errors and throw it once at the end:

https://stackoverflow.com/a/30118121/196405

Reply:
Great idea.  I will implement that.  Thanks for the style suggestion.

Reply:
Since this is a common, reusable pattern, it might be possible to put it into a reusable static method like:

```java
public static <T> void forEachCatchRTEs(Stream<T> stream, Consumer<T> consumer) {
  stream.flatMap(o -> {
￼       try {
         consume(o);
￼         return null;
￼       } catch (RuntimeException e) {
￼         return Stream.of(e);
￼       }
￼     }).reduce((e1, e2) -> {
￼           e1.addSuppressed(e2);
￼           return e1;
￼         }).ifPresent(e -> {
￼           throw e;
￼         });
}
```

Then, each time you need it:

```java
forEachCatchRTEs(pathsToMonitor.stream(), o -> { addFileToMonitor(o); log.debug("monitoring {}", o); });
```

Reply:
Thanks for the good idea.   I will put that function in now.